src/cow-react/modules/limitOrders/containers/SettingsWidget/index.tsx

```diff
@@ -18,21 +18,25 @@ export function SettingsWidget() {
 
   const onStateChanged = useCallback(
     (state: Partial<LimitOrdersSettingsState>) => {
-      const isExpertModeOn = !settingsState.expertMode && state.expertMode
-      const isExpertModeOff = settingsState.expertMode && !state.expertMode
+      const { expertMode } = state
 
-      if (isExpertModeOn) {
-        setShowExpertConfirm(true)
-      } else if (isExpertModeOff) {
-        updateSettingsState({ expertMode: false, showRecipient: false })
+      if (expertMode !== undefined) {
+        const isExpertModeOn = !settingsState.expertMode && expertMode
+        const isExpertModeOff = settingsState.expertMode && !expertMode
```

Comment:
`isExpertModeOff = !isExpertModeOn`
😄

Reply:
This logic was a bit crazy 

### isExpertModeOff = !isExpertModeOn?

```
const isExpertModeOn = !settingsState.expertMode && expertMode
isExpertModeOff = !isExpertModeOn
isExpertModeOff = !(!settingsState.expertMode && expertMode)

isExpertModeOff = settingsState.expertMode || !expertMode
```

Since `settingsState.expertMode && !expertMode <> settingsState.expertMode || !expertMode`

Then shouldn't be the same